services/datamanager/builtin/builtin.go

```diff
@@ -129,28 +129,33 @@ func readyToSync(ctx context.Context, s selectiveSyncer, logger logging.Logger)
 // builtIn initializes and orchestrates data capture collectors for registered component/methods.
 type builtIn struct {
 	resource.Named
-	logger                 logging.Logger
-	captureDir             string
-	captureDisabled        bool
-	collectorsMu           sync.Mutex
-	collectors             map[resourceMethodMetadata]*collectorAndConfig
-	lock                   sync.Mutex
-	backgroundWorkers      sync.WaitGroup
-	fileLastModifiedMillis int
-
-	additionalSyncPaths []string
-	tags                []string
-	syncDisabled        bool
-	syncIntervalMins    float64
-	syncRoutineCancelFn context.CancelFunc
-	syncer              datasync.Manager
-	syncerConstructor   datasync.ManagerConstructor
-	filesToSync         chan string
-	maxSyncThreads      int
-	cloudConnSvc        cloud.ConnectionService
-	cloudConn           rpc.ClientConn
-	syncTicker          *clk.Ticker
-	maxCaptureFileSize  int64
+	closedCtx                 context.Context
+	closedCancelFn            context.CancelFunc
+	logger                    logging.Logger
+	captureDir                string
+	captureDisabled           bool
+	collectorsMu              sync.Mutex
+	collectors                map[resourceMethodMetadata]*collectorAndConfig
+	lock                      sync.Mutex
+	backgroundWorkers         sync.WaitGroup
+	propagateDataSyncConfigWG sync.WaitGroup
+	fileLastModifiedMillis    int
+
+	additionalSyncPaths          []string
+	tags                         []string
+	syncDisabled                 bool
+	syncIntervalMins             float64
+	syncRoutineCancelFn          context.CancelFunc
+	syncer                       datasync.Manager
+	syncerConstructor            datasync.ManagerConstructor
+	filesToSync                  chan string
+	maxSyncThreads               int
+	syncConfigUpdated            bool
+	syncerNeedsToBeReInitialized bool
```

Comment:
I added `syncerNeedsToBeReInitialized`, `syncConfigUpdated`, `closedCtx`, `closedCancelFn`, and `propagateDataSyncConfigWG`.